tests/unit/lowres_clock_test.cc

```diff
@@ -52,6 +52,30 @@ SEASTAR_TEST_CASE(steady_clock_sanity) {
     });
 }
 
+//
+// Sanity check the accuracy of the steady low-resolution clock in debug mode.
+//
+SEASTAR_TEST_CASE(steady_clock_sanity_preempt) {
+    return do_with(lowres_clock::now(), [](auto &&t1) {
+        // Sleep duration must be higher than the task quota to ensure that
+        // preemption is requested before we yield.
+        static constexpr auto sleep_duration = std::chrono::milliseconds(100);
+        std::this_thread::sleep_for(sleep_duration);
+
+        // Yield to the reactor to give it a chance to update the
+        // low-resolution clock. Yield just schedules an empty task and
+        // waits for it to be executed. There is nothing special about it.
+        return ::seastar::yield().then([&t1] {
+            auto const elapsed = lowres_clock::now() - t1;
+            auto const minimum_elapsed = 0.9 * sleep_duration;
+
+            BOOST_REQUIRE(elapsed >= minimum_elapsed);
+
+            return make_ready_future<>();
+        });
+    });
+}
```

Comment:
Please write the test as a coroutine (or a Seastar thread), now that C++20 and above is mandatory there's no reason for introducing new code in continuation style.

Reply:
Fixed.